[PySpark] - Add extra str functions to pyspark api

mariotaddeucci opened the pull request.

mariotaddeucci committed:
Add extra str functions to pyspark api

binste commented:
Great to see this PR, thanks!

For the reviewers, just to note that at the same time, I've added `substr` in #14863. I can remove mine in that PR or we can remove it here, doesn't matter to me.

mariotaddeucci committed:
Merge branch 'duckdb:main' into pyspark-str-functions

mariotaddeucci committed:
[PySpark] Add string functions: substring, contains, reverse, concat and instr

mariotaddeucci commented:
> Great to see this PR, thanks!
> 
> For the reviewers, just to note that at the same time, I've added `substr` in #14863. I can remove mine in that PR or we can remove it here, doesn't matter to me.

Removed "substr".

Mytherin commented:
Thanks!
